functions/Disable-DbaHideInstance.ps1

```diff
@@ -0,0 +1,137 @@
+
+    .DESCRIPTION
+        Disables the Hide Instance setting of the SQL Server network configuration.
+
+        This requires access to the Windows Server and not the SQL Server instance. The setting is found in SQL Server Configuration Manager under the properties of SQL Server Network Configuration > Protocols for "InstanceName".
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances.
+
+    .PARAMETER Credential
+        Allows you to login to the computer (not SQL Server instance) using alternative Windows credentials
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: HideInstance, Security
+        Author: Gareth Newman (@gazeranco), ifexists.blog
+
+        Website: https://dbatools.io
+        Copyright: (c) 2019 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .EXAMPLE
+        PS C:\> Disable-DbaHideInstance
+
+        Disables Hide Instance of SQL Engine on the default (MSSQLSERVER) instance on localhost. Requires (and checks for) RunAs admin.
+
+    .EXAMPLE
+        PS C:\> Disable-DbaHideInstance -SqlInstance sql01\SQL2008R2SP2
+
+        Disables Hide Instance of SQL Engine for the SQL2008R2SP2 on sql01. Uses Windows Credentials to both connect and modify the registry.
+
+    .EXAMPLE
+        PS C:\> Disable-DbaHideInstance -SqlInstance sql01\SQL2008R2SP2 -WhatIf
+
+        Shows what would happen if the command were executed.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Low", DefaultParameterSetName = 'Default')]
+    param (
+        [Parameter(ValueFromPipeline)]
+        [DbaInstanceParameter[]]
+        [DbaInstanceParameter[]]$SqlInstance = $env:COMPUTERNAME,
+        [PSCredential]$Credential,
+        [switch]$EnableException
+    )
+    process {
+
+        foreach ($instance in $SqlInstance) {
+            Write-Message -Level VeryVerbose -Message "Processing $instance." -Target $instance
+            if ($instance.IsLocalHost) {
+                $null = Test-ElevationRequirement -ComputerName $instance -Continue
+            }
+
+            try {
+                Write-Message -Level Verbose -Message "Resolving hostname."
```

Comment:
We can go ahead and remove this verbose statement because the Resolve-DbaNetworkName has verbose output as well that the user will see what is going on.

Reply:
So this block along with the following if login on lines 75-77 need to be combined...the use of the `Turbo` parameter is just a shortcut version to try and do one last check to resolve the server name.

```powershell
try {
    $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -EnableException
} catch {
    try {
        $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -Turbo -EnableException
    } catch {
        Stop-Function -Message "Issue resolving $instance" -Target $instance -Category InvalidArgument -Continue
    }
}
```